Drive/rover_drive_system.hpp

```diff
@@ -150,19 +150,104 @@ class RoverDriveSystem
   /// @param speed the new movement speed of the rover
   void SetWheelSpeed(units::angular_velocity::revolutions_per_minute_t speed)
   {
-    // TODO: Implement linear interpolation (exponential moving average)
+    //Implement linear interpolation (exponential moving average)
     try
     {
-      left_wheel_.SetHubSpeed(speed);
-      right_wheel_.SetHubSpeed(speed);
-      back_wheel_.SetHubSpeed(speed);
-    }
+      sjsu::LogInfo("made it to SetWheelSpeed()");
+       
+       auto clampedHubSpeed = std::clamp(speed, kMaxNegSpeed, kMaxPosSpeed);
+      //static cast to use clampedHubSpeed in lerp function
+      long double new_speed = static_cast<long double>(clampedHubSpeed);
+      //intialize the variable we will use to store the revolutions_per_minute_t that is returned on each call to std::lerp
+      long double lerpSpeed_leftWheel = 0;
+      long double lerpSpeed_rightWheel = 0;
+      long double lerpSpeed_backWheel = 0;
+      //intialize the variable we will use to store the current speed of the motor after each call to lerp
+      long double currentSpeed_leftWheel = 0;
+      long double currentSpeed_rightWheel = 0;
+      long double currentSpeed_backWheel = 0;
+
+      //want to make sure we are supposed to increase or decrease the speed when setting the speed
+      bool increaseSpeed_leftWheel = true;
+      bool increaseSpeed_rightWheel = true;
+      bool increaseSpeed_backWheel = true;
+      if(left_wheel_.RequestFeedbackFromMotor().GetFeedback().speed > speed)
+      {
+        increaseSpeed_leftWheel = false;
+      }
+      if(right_wheel_.RequestFeedbackFromMotor().GetFeedback().speed > speed)
+      {
+        increaseSpeed_rightWheel = false;
+      }
+      if(back_wheel_.RequestFeedbackFromMotor().GetFeedback().speed > speed)
+      {
+        increaseSpeed_backWheel = false;
+      }
```

Comment:
We don't actually need to ask the motors for their current speed for this. We don't need true speed. The motors will handle getting to the speed a the rate they are programmed to. We just need to send them closer and closer steps to the goal. The won't be aligned but its also not necessary to be perfectly aligned with the output when it comes to lerp.